### PR TITLE
linuxPackages.evdi: mark broken on linux 6.6

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

ZHF: https://github.com/NixOS/nixpkgs/issues/265948

Fixes Hydra failures:

* https://hydra.nixos.org/build/239845714
  * See issue https://github.com/NixOS/nixpkgs/issues/265868 reported by @Infinidoge (thanks!)
  * `evdi` kernel module is incompatible with Linux 6.6, so mark it as broken
* https://hydra.nixos.org/build/239848621
  * Fix the build for Linux 4.19

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/evdi/default.nix [hunk 1]
```diff
@@ -16,7 +16,11 @@ stdenv.mkDerivation rec {
     hash = "sha256-em3Y56saB7K3Wr31Y0boc38xGb57gdveN0Cstgy8y20=";
   };
 
-  env.NIX_CFLAGS_COMPILE = "-Wno-error -Wno-error=sign-compare";
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-Wno-error"
+    "-Wno-error=discarded-qualifiers" # for Linux 4.19 compatibility
+    "-Wno-error=sign-compare"
+  ];
```
Comment (alyssais) [inline pkgs/os-specific/linux/evdi/default.nix:23]: This seems like an unrelated change?

Comment (anthonyroussel) [reply pkgs/os-specific/linux/evdi/default.nix:23]: Thanks for the review @alyssais 
Indeed I went too fast, I splitted this ZHF failure for Linux 4.19 in another commit.
Is it okay for you?

Comment (alyssais) [reply pkgs/os-specific/linux/evdi/default.nix:23]: That's much better, thanks.

**Review by alyssais — APPROVED** [review]

**Review by Infinidoge — APPROVED** [review]
